digitalfabrik/entitlementcard
# 1206 Notes for applications

### Short description

As a service user i might want to add notes to the current application, f.e. how often the applicant was contacted or sth similar.

### Proposed changes



- add `updateNote` fkt in the backend
- adjust database schema and ApplicationView for applications
- add database migration for applications table
- add reusable `TextAreaDialog` component
- add `NotesDialogController` for data handling
- fix loading spinner position to avoid container jumping while loading data
- limited notes to 1000 chars
- show quick note by hovering "Note anzeigen"
- add test

### Resolved issues



Fixes: #1206

### Testing
- run migrate db (dev only)
- open applications
- select an application
- click on 'Notiz anzeigen'
- add text and save -> text should be saved on open
- add text and close without saving (or using backdrop, click on overlay to close) -> text updates shouldn't be saved
- reach the character limit and check if note can be saved
- test if tooltip appears

Note: I slightly adjusted the design, since Button colors were wrong in the design and used the standard `DialogFooter` from blueprint
